Domybest (#1)
Kosuke-iga/SSS-master-thesis: doc/3_previous.tex

\chapter{Previous Scheme}\markright{Previous Scheme}\label{sec:previous_scheme}
\section{Overview of the previous scheme} 
In order to deal with spreading IoT malwares with device-side detections in more practical means, CNN is utilized in the previous scheme \cite{previous}.
The main idea of the previous scheme is that there exists the difference of GSIs converted from each malware Binary Data (BD) which is always acquired in IoT devices due to the exectable format of them.
In that scheme \cite{previous}, a malware binary is reformatted as an 8-bit sequence and then be converted to a GSI which has one channel and pixel values from 0 to 255.

> Some examples of malware and benign software GSIs are shown by ず挿入.

Reviewer: ```suggestion
\figurename~\ref{} shows the examples of malware and benign software GSIs.

```
基本的に最初に"Fig. ? shows hoge hoge. " って言った後に、"As  shown in Fig. ?, "とかで説明するようにした方が読みやすいよ。

> By comparison, some novel differences between them can be observed even by human eyes.

Reviewer: ```suggestion
As shown in \figurename~\ref{}. , some novel differences between them can be observed even by human eyes.

```

For example, it can be seen that malware GSIs always are more dense.
On the other hand, the GSI of benign softwares tend to have larger header parts than malwares.

These GSIs can represents each features and be fed into CNN.
Since, CNN can extract effective features from these GSI automatically by learning deep non-linear features even that can be hardly descovered and understood by human eyes.
Furthermore, once trained, the network itself is lightweight and can be run with tiny computational resources, since only the trained parameters and information of network structure are kept.
Thus, CNN is a suitable scheme for IoT malware detections since it does not demand feature engineering and also computational costs while the detection.

\section{Shortcoming of the previous scheme } 
Although the previous scheme can cope with spreading IoT malwares with device-side detection in more practical means, it is susceptible to evasive techniques against the inputs of the CNN, GSIs and binary.
Adversaries can evade the CNN based detection with two evasive techniques against each input as shown in 図.

> Malwares whose GSIs are manipulated to evade the detection are called Adversarial Malware (AM) and also malwares whose binary is manipulated are called Obfuscated Malware (OM) in this research.  

Reviewer: ```suggestion
Malware samples whose GSIs are manipulated to evade the detection are called Adversarial Malware (AM), and also malware samples whose binary codes are obfuscated by packing tools are called Obfuscated Malware (OM) in this thesis.  

```
両方manipulated だとわかりにくいから、パッキングツール（または圧縮技術）で難読化されたものって言っていいと思う。
